core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -651,60 +651,126 @@ class UserService(protected val ctx: RawlsRequestContext,
-              } else {
-                Future.failed(regrets)
-              }
-            }
+            Seq(getV2BillingPolicy(projectAccessUpdate.role))
         }
+        _ <- addUserToBillingProjectInner(projectName, projectAccessUpdate, policies)
       } yield {}
     }
 
+  def getBillingProfileId(projectName: RawlsBillingProjectName): Future[Option[String]] =
+    dataSource.inTransaction { dataAccess =>
+      dataAccess.rawlsBillingProjectQuery.load(projectName).map { billingProjectOpt =>
+        billingProjectOpt
+          .getOrElse(
+            throw new RawlsException(s"Billing Project ${projectName.value} does not exist in Rawls database")
+          )
+          .billingProfileId
+      }
+    }
+
   def removeUserFromBillingProject(projectName: RawlsBillingProjectName,
                                    projectAccessUpdate: ProjectAccessUpdate
   ): Future[Unit] =
     requireProjectAction(projectName, SamBillingProjectActions.alterPolicies) {
-      val policy = projectAccessUpdate.role match {
-        case ProjectRoles.Owner => SamBillingProjectPolicyNames.owner
-        case ProjectRoles.User  => SamBillingProjectPolicyNames.workspaceCreator
-      }
+      removeUserFromBillingProjectInner(projectName, projectAccessUpdate)
+    }
+
+  private def removeUserFromBillingProjectInner(projectName: RawlsBillingProjectName,
+                                                projectAccessUpdate: ProjectAccessUpdate
+  ): Future[Unit] =
+    for {
+      _ <- samDAO
+        .removeUserFromPolicy(SamResourceTypeNames.billingProject,
+                              projectName.value,
+                              getV2BillingPolicy(projectAccessUpdate.role),
+                              projectAccessUpdate.email,
+                              ctx.userInfo
+        )
+        .recover {
+          case e: RawlsExceptionWithErrorReport if e.errorReport.statusCode.contains(StatusCodes.BadRequest) =>
+            throw new RawlsExceptionWithErrorReport(e.errorReport.copy(statusCode = Some(StatusCodes.NotFound)))
+        }
+    } yield {}
 
+  def removeUserFromBillingProjectV2(projectName: RawlsBillingProjectName,
+                                     projectAccessUpdate: ProjectAccessUpdate
+  ): Future[Unit] =
+    requireProjectAction(projectName, SamBillingProjectActions.alterPolicies) {
       for {
-        _ <- samDAO
-          .removeUserFromPolicy(SamResourceTypeNames.billingProject,
-                                projectName.value,
-                                policy,
-                                projectAccessUpdate.email,
-                                ctx.userInfo
-          )
-          .recover {
-            case e: RawlsExceptionWithErrorReport if e.errorReport.statusCode.contains(StatusCodes.BadRequest) =>
-              throw new RawlsExceptionWithErrorReport(e.errorReport.copy(statusCode = Some(StatusCodes.NotFound)))
-          }
+        billingProjectId <- getBillingProfileId(projectName)
```

Comment:
I think this var should be called `billingProfileId`

Reply:
Definitely, thanks.